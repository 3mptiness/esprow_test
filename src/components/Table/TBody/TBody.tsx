import React, { FC, memo, PropsWithChildren } from 'react';

type TBodyProps = {
	scroll: number;
	fullHeight: number;
	rowHeight: number;
	rowsLength: number;
};

export const TBody: FC<PropsWithChildren<TBodyProps>> = memo((props) => {
	const {
		scroll, fullHeight, children, rowHeight, rowsLength,
	} = props;

	return (
		<tbody>
			<tr style={{ height: scroll }} />
			{children}
			<tr style={{ height: fullHeight - scroll - rowsLength * rowHeight }} />
		</tbody>
	);
});
