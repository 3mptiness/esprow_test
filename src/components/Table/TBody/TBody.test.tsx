import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { TBody } from './TBody';

describe('TBody', () => {
	it('success TBody render', () => {
		expect(
			createRenderer().render(
				<TBody scroll={10} fullHeight={100} rowHeight={10} rowsLength={5} />
			)
		).toMatchSnapshot();
	});
});
