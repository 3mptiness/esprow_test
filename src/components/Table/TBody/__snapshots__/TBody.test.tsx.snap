// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TBody success TBody render 1`] = `
<tbody>
  <tr
    style={
      Object {
        "height": 10,
      }
    }
  />
  <tr
    style={
      Object {
        "height": 40,
      }
    }
  />
</tbody>
`;
